File: transpose_matrix.py
# Easy python solution
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        return [list(row) for row in zip(*matrix)]    
      
# Normal solution
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        new_num_rows = len(matrix[0])
        new_matrix = [[] for i in range(new_num_rows)]
        
        for i in range(len(matrix)):
            for j in range(new_num_rows):
                new_matrix[j].append(matrix[i][j])
        return new_matrix
    

File: transpose_matrix.js
// https://leetcode.com/problems/transpose-matrix/

var transpose = function(matrix) {
    let new_num_rows = matrix[0].length;
    let new_matrix = [];

    for (let i = 0; i < matrix[0].length; i++) {
        let temp_row = [];
        for (let j = 0; j < matrix.length; j++) {
            temp_row.push(matrix[j][i])
        }
        new_matrix.push(temp_row);
    }
    return new_matrix;
    
};
